nowcasting_dataset/data_sources/gsp/gsp_model.py

```diff
@@ -21,7 +21,8 @@ class GSP(DataSourceOutput):
     @classmethod
     def model_validation(cls, v):
         """Check that all values are non NaNs"""
-
+        # Fill NaNs with Zeros
+        v["power_mw"] = v.check_nan_and_fill_warning(data=v.power_mw)
```

Comment:
im not sure this should be the default, this could lead to us fillng a load of zeros when there shouldnt be

Reply:
Okay, should I make it an option? I'm not sure the best way of telling when we should fill in NaNs or when we shouldn't be doing so. When I was filling in NaNs with zeros in get_example, NaNs would still show up in validation. But I also could just remove this NaN to zero and have the PR just be the finding the closest GSP? IT could be this issue is mostly because of changes between when v15 was created and now?